src/azure-cli-telemetry/azure/cli/telemetry/util.py

```diff
@@ -0,0 +1,77 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import stat
+import logging
+import logging.handlers
+from datetime import datetime
+
+from azure.cli.telemetry.const import TELEMETRY_NOTE_NAME, MANDATORY_WAIT_PERIOD
+
+
+def should_upload(config_dir):
+    """Returns True if it is the right moment to add telemetry.
+    Before the client request a telemetry add process, it can invoke this methods to test if it is the right moment.
+    The conditions are:
+        1. The telemetry.txt file doesn't exist; OR
+        2. The telemetry.txt file is a regular file AND there have been enough time passed since last add.
+    """
+    logger = logging.getLogger('telemetry.check')
+
+    telemetry_note_path = os.path.join(config_dir, TELEMETRY_NOTE_NAME)
+    if not os.path.exists(telemetry_note_path):
+        logger.info('Positive: The %s does not exist.', telemetry_note_path)
+        return True
+
+    file_stat = os.stat(telemetry_note_path)
+    if not stat.S_ISREG(file_stat.st_mode):
```

Comment:
Unfortunately yes. There is no meaningful recovery here because we won't be able to know if the user intentionally does that or not. If it is a user-created folder to occupied the exact file, we should not remove or alter it. I will side on the safe-side to give up telemetry.

 On the other hand, there isn't much root for this file to be corrupted.